app/controllers/application_controller.rb

```diff
@@ -1,20 +1,31 @@
 class ApplicationController < ActionController::Base
   include UsersHelper
   include SessionsHelper
-
-  def redirect
-    # if params[:preview] == 1 && current_user&.admin?
-    if current_user&.admin? && params[:preview].nil?
-      redirect_to admin_path
-    else
-      true
-    end
-  end
+  include OrdersHelper
 
   def authorize_user!
     unless logged_in?
       flash.now[:danger] = 'You have to log in'
       redirect_to root_url
     end
   end
+
+  def logged_in_user
+    unless logged_in?
+      store_location
+      flash[:danger] = 'Please log in.'
+      redirect_to login_url
+    end
+  end
+
+  def correct_user
```

Comment:
I'd try to rename this method, same for `correct_order`. Maybe use `authenticate_user` because it's an authentication check?